### PR TITLE
#86 staking and unstaking

### PR DESCRIPTION
**Pull request by memasdeligeorgakis**
# What does this PR do
It adds staking and unstaking views and the state management for them. The intention is to be very plain. It contains a naive field validation that should be replaced with a central one in the utils.

image

*initial view, confirm disabled*

image

*valid staking amount*

image

*valid unstaking amount*

image

*invalid amount*

TODO:
* likely the modal should be hugging the content while having a max height. Now it takes a constant height even if the content would not require so much.

### CODE REVIEW

File: apps/namada-interface/src/App/Staking/Staking.tsx [hunk 1]
```diff
@@ -68,9 +99,11 @@ export const Staking = (props: Props): JSX.Element => {
   } = props;
 
   // these 2 are needed for validator details
-  const stakingPositionsWithSelectedValidator = myStakingPositions.filter(
-    (validator) => validator.validatorId === selectedValidatorId
-  );
+  const stakingPositionsWithSelectedValidator =
+    myStakingPositions &&
```
Comment (mateuszjasiuk) [inline apps/namada-interface/src/App/Staking/Staking.tsx:103]: `myStakingPosition` is of type `MyStakingPosition[]`, so either `myStakingPosition` is always true or the type has to change. Kinda weird that typescript does not show it as "always true"

**Review by mateuszjasiuk — APPROVED** [review]
Looks good to me! \
Left one comment regarding type.

Also not connected to this PR directly, when I was looking at redux state:
`stakingOrUnstakingState` shows `0 | 1 | 2`. Maybe it would be nice to change enum to have string values, so it's more clear when looking at state :)

File: apps/namada-interface/src/App/Staking/Staking.tsx (hunk 1 above)
Comment (memasdeligeorgakis) [reply apps/namada-interface/src/App/Staking/Staking.tsx]: Odd that it was there, but now I removed it and TS seems to be happy about it as expected.

File: apps/namada-interface/src/slices/README.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# The state management of namada-interface
+
+## Description
+This directory contains the state management for the application. It is divided by main topics (staking and governance, transfers, ...). Each directory contains the following files:
+* **actions.ts** - this contains all the actions for the state. All the side effects (network calls, calls to libs, ...) are being performed from here.
+* **reducers.ts** - all reducers of the state are here. This is where the state changes are being committed. Commonly all actions have 3 reducers associated with them:
+    * pending - this is being triggered with the initial dispatch call (mostly originating from the UI or view Components). The payload to this is same as the action was dispatched with.
```
Comment (jurevans) [inline apps/namada-interface/src/slices/README.md:7]: Very minor suggestion: This should likely be `**pending**` to match the others

**Review by jurevans — APPROVED** [review]
Looks good! Just one minor comment in the slices README

**Pull request merged**